app/scripts/lib/notification-manager.js

```diff
@@ -106,7 +106,7 @@ export default class NotificationManager extends EventEmitter {
     return windows
       ? windows.find((win) => {
           // Returns notification popup
-          return win && win.type === 'popup' && win.id === this._popupId;
```

Comment:
Can you explain why this works as a fix and what the original check was meant for? Are there any ramifications of removing this check?

Reply:
Thanks for checking ~ it was the issue due to refresh changed `this._popupId` to undefined so we cannot find a match for current window opened. And you are right there would be some regressions, for instance, if there's a popup window from another extension, removing this will hijack another window. Therefore the final solution will be saving this id in the appState level.